Gems/ROS2/Code/Source/ContactSensor/ROS2ContactSensorComponent.cpp

```diff
@@ -0,0 +1,203 @@
+
+    void ROS2ContactSensorComponent::FrequencyTick()
+    {
+        // Connects the collision handlers if not already connected
+        AzPhysics::SystemInterface* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get();
+        AZ_Assert(physicsSystem, "No physics system.");
+
+        AzPhysics::SceneInterface* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        AZ_Assert(sceneInterface, "No scene interface.");
+
+        AzPhysics::SceneHandle defaultSceneHandle = sceneInterface->GetSceneHandle(AzPhysics::DefaultPhysicsSceneName);
+        AZ_Assert(defaultSceneHandle != AzPhysics::InvalidSceneHandle, "Invalid default physics scene handle.");
+
+        if (!m_onCollisionBeginHandler.IsConnected() || !m_onCollisionPersistHandler.IsConnected() ||
+            !m_onCollisionEndHandler.IsConnected())
+        {
+            if (auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get())
+            {
+                AZStd::pair<AzPhysics::SceneHandle, AzPhysics::SimulatedBodyHandle> foundBody =
+                    physicsSystem->FindAttachedBodyHandleFromEntityId(GetEntityId());
+                AZ_Warning("Contact Sensor", foundBody.first != AzPhysics::InvalidSceneHandle, "Invalid scene handle");
+                if (foundBody.first != AzPhysics::InvalidSceneHandle)
+                {
+                    AzPhysics::SimulatedBodyEvents::RegisterOnCollisionBeginHandler(
+                        foundBody.first, foundBody.second, m_onCollisionBeginHandler);
+                    AzPhysics::SimulatedBodyEvents::RegisterOnCollisionPersistHandler(
+                        foundBody.first, foundBody.second, m_onCollisionPersistHandler);
+                    AzPhysics::SimulatedBodyEvents::RegisterOnCollisionEndHandler(
+                        foundBody.first, foundBody.second, m_onCollisionEndHandler);
+                }
+            }
+        }
+
+        // Publishes all contacts
+        gazebo_msgs::msg::ContactsState msg;
+        auto* ros2Frame = Utils::GetGameOrEditorComponent<ROS2FrameComponent>(GetEntity());
+        msg.header.frame_id = ros2Frame->GetFrameID().data();
+        msg.header.stamp = ROS2Interface::Get()->GetROSTimestamp();
+
+        {
+            AZStd::lock_guard<AZStd::mutex> lock(m_activeContactsMutex);
+            for (auto& [id, contact] : m_activeContacts)
+            {
+                msg.states.push_back(contact);
+            }
+        }
+
+        m_contactsPublisher->publish(AZStd::move(msg));
+    }
+
+    void ROS2ContactSensorComponent::AddNewContact(const AzPhysics::CollisionEvent& event)
+    {
+        AZ::Entity* contactedEntity = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(
+            contactedEntity, &AZ::ComponentApplicationRequests::FindEntity, event.m_body2->GetEntityId());
+        gazebo_msgs::msg::ContactState state;
+        AZ_Assert(contactedEntity != nullptr, "Invalid entity pointer value") state.collision1_name =
```

Comment:
Code seems to not formatted. 
AZ_assert will `explode` when the statement is false.